src/java/voldemort/store/readonly/ReadOnlyFileDiffer.java

```diff
@@ -0,0 +1,383 @@
+package voldemort.store.readonly;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Iterator;
+import org.apache.commons.cli.BasicParser;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import voldemort.utils.ByteUtils;
+import voldemort.utils.EventThrottler;
+import voldemort.utils.Pair;
+
+
+/**
+ * ReadOnlyFileDiffer is used to compare between different versions of specific store and give the number of how many
+ * records are added/deleted/updated/same.
+ *
+ * N.B.: This tool only works with the Read-Only V2 format.
```

Comment:
Nitpick, since you are inside a JavaDoc block, you can use the following notation:

{@link voldemort.store.readonly.ReadOnlyStorageFormat#READONLY_V2}

This creates a link in auto-generated JavaDoc and also allows you to follow the link using IntelliJ. Not really important, but I just thought I'd mention it (: